docs/FORD.OpenATF.SDD.TPL.md

```diff
@@ -0,0 +1,229 @@
+  - Data types from chapter 4.2
+  Note: See modules/schema_validation.lua as an example
+5. Remove this instruction explanation from beginning to {END_INSTRUCTION} tags
+
+General notes/reminders:
+- Look at "modules/docs/Validation SW Detail Design.md" example
+- Commit both: images and them source to the git repository
+- SDD file extension shall be 'md' 
+- the preferable path for SDD is modules/COMPONENT/docs
+- the preferable path for SDD images and there source files is modules/COMPONENT/docs/assets
+- Foe adding lua file to documentation start this lua file with a following ldoc comment ('---')
+- For adding images in MD format follow https://www.stack.nl/~dimitri/doxygen/manual/markdown.html#md_images
+- As a tool for image preparing could be used Gliffy digram
+      https://adc.luxoft.com/confluence/pages/createpage.action?showGliffyMacro=true&fromCreateDialog=true&spaceKey=APPLINK
+    OR plantuml diagram
+      http://plantuml.com/state.html
+- empty string requires before the lists for correct parsing by ldoc
+
+Known issues:
+- Markdown tables are not working properly in md files
+
+For more information, please follow:
+- LDoc documentation
+    https://stevedonovan.github.io/ldoc/manual/doc.md.html
+- Markdown sintaxyx by doxygen
+    http://www.stack.nl/~dimitri/doxygen/manual/markdown.html
+- Text-base UML tool
+    http://plantuml.com/
+- Article "Providing design documentation with code changes"
+    https://github.com/smartdevicelink/sdl_core/wiki/Providing-design-documentation-with-code-changes
+-------{END_INSTRUCTION}-------
+
+# <!Component Name> SW Detail Design
+
+## 1 Introduction
+The document is intended to support software developers,
+maintenance and integration engineers with sufficient,
+detailed information concerning the design, development and
+deployment concepts, to accomplish their respective tasks without reliance on the authors.
+
+## 1.1 Rationale
+<!Component Name> implements ATF Architectural Solution according to [ATF SW Architecture](https://smartdevicelink.com/en/guides/pull_request/93dee199f30303b4b26ec9a852c1f5261ff0735d/atf/components-view/)
+
+## 1.2 Scope
+<!Component Name> extracted as a separate component for:
+
+<!!!!!!!!!!!BEGIN_INSTRUCTION!!!!!!!!!!!!!>
+Here need to be added a reason and short description of the components functionality
+Example:
+  Security Manager component extracted as a separate module for
+  Ford channel data protection.
+  This components is used to :
+  - Provide security communications
+  - Protect income and outcome business layer data from interception
+  - Verify the relation between a mobile application certificate and its owner
+<!!!!!!!!!!!END_INSTRUCTION!!!!!!!!!!!!!>
+
+## 1.3 Abbreviations and definitions
+
+<!!!!!!!!!!!BEGIN_INSTRUCTION!!!!!!!!!!!!!>
+Here need to be added all component-specific terms, as
+*RPC* - Remote Procedure Call.
+<!!!!!!!!!!!END_INSTRUCTION!!!!!!!!!!!!!>
+
+## 2. Component detail design
+## 2.1 Design solutions
```

Comment:
@LuxoftAKutsan,
Yes, all chapters are header2 for correct showing them in the `Contents` menu.
The first header is a name and H1.